src/vos/vos_pool.c

```diff
@@ -105,13 +105,17 @@ pool_hop_free(struct d_ulink *hlink)
 
 	if (pool->vp_io_ctxt != NULL) {
 		rc = bio_ioctxt_close(pool->vp_io_ctxt);
-		if (rc)
-			D_ERROR("Closing VOS I/O context:%p pool:"DF_UUID"\n",
-				pool->vp_io_ctxt, DP_UUID(pool->vp_id));
-		else
+		if (rc) {
+			D_ERROR("Closing VOS I/O context:%p pool:"DF_UUID" : "
+				DF_RC"\n", pool->vp_io_ctxt,
+				DP_UUID(pool->vp_id), DP_RC(rc));
+			/* Free the I/O context no matter if close succeeded */
+			D_FREE(pool->vp_io_ctxt);
```

Comment:
I'm wondering why we have to free it when bio_ioctxt_close() failed? Just to make Coverity happy? If so, I think we'd do it in bio_ioctxt_close() (to delete it from list and free it without bio_blob_close()).

Reply:
Hmm. We free the pool structure at the end and this memory will be lost. Or I don't understand the logic.